Analysis/RecoMCTruthLink/src/RecoMCTruthLinker.cc

```diff
@@ -1175,18 +1175,20 @@ void RecoMCTruthLinker::clusterLinker(  LCEvent * evt,  LCCollection* mcpCol ,
                                 << " contributors " << std::endl;
 
         nsimhit++;
-        for(int j=0;j<simHit->getNMCContributions() ;j++){
-          
-          MCParticle* mcp = simHit->getParticleCont( j ) ;
-          double e  = simHit->getEnergyCont( j ) * calib_factor ;
-          streamlog_out( DEBUG3 ) <<"     true contributor = "<< mcp->id() << " e: " << e 
-                                  <<" mapped to " <<remap_as_you_go.find(mcp)->second->id() << std::endl;
+        for(int j=0; j<simHit->getNMCContributions(); j++){          
+          MCParticle* mcp = simHit->getParticleCont(j);
           if ( mcp == 0 ) {
             streamlog_out( DEBUG7 ) <<"      simhit = "<< simHit << " has no creator " <<std::endl;
-            // streamlog_out( DEBUG7 ) <<"         true contributor = "<< mcp << " e: " << e 
-            //    <<" mapped to " <<remap_as_you_go.find(mcp)->second << std::endl;
-            continue ; 
+            continue;
           }
+          if ( remap_as_you_go.find(mcp) == remap_as_you_go.end() ){
+            streamlog_out( DEBUG7 ) <<"      simhit = "<< simHit << "; its true contributor "<<mcp->id()<<" is not found in the map"<<std::endl;
+            continue;
+          }
+
+          double e  = simHit->getEnergyCont(j) * calib_factor;
+          streamlog_out( DEBUG3 ) <<"     true contributor = "<< mcp->id() << " e: " << e 
+                                  <<" mapped to " <<remap_as_you_go.find(mcp)->second->id() << std::endl;
```

Comment:
There is a double lookup in the map here, once for the `if` condition and once in the debug statement. Storing the `find` result in a variable and re-using that would save one potentially rather expensive map lookup.